### PR TITLE
feat(enumerables): optimize array mapping by using Array.Clone()

### PR DESCRIPTION
**Pull request by latonz**

### CODE REVIEW

File: src/Riok.Mapperly/Emit/SyntaxFactoryHelper.cs [hunk 1]
```diff
@@ -141,17 +150,51 @@ public static InvocationExpressionSyntax StaticInvocation(IMethodSymbol method,
...
-    public static LocalDeclarationStatementSyntax DeclareVariable(string variableName, ExpressionSyntax initializationValue)
+    public static BinaryExpressionSyntax LessThan(ExpressionSyntax left, ExpressionSyntax right)
+    {
+        return BinaryExpression(
+            SyntaxKind.LessThanExpression,
+            left,
+            right);
+    }
+
+    public static ForStatementSyntax ForCounterLoop(string counterName, ExpressionSyntax upperExclusiveBound, StatementSyntax body)
```
Comment (CommonGuy) [inline src/Riok.Mapperly/Emit/SyntaxFactoryHelper.cs:161]: Is this used anywhere?

File: test/Riok.Mapperly.Tests/Mapping/EnumerableTest.cs [hunk 2]
```diff
@@ -10,7 +10,7 @@ public void ArrayToArrayOfPrimitiveTypes()
             "int[]");
         TestHelper.GenerateSingleMapperMethodBody(source)
             .Should()
-            .Be("return System.Linq.Enumerable.ToArray(source);");
+            .Be("return (int[])source.Clone();");
```
Comment (CommonGuy) [inline test/Riok.Mapperly.Tests/Mapping/EnumerableTest.cs:13]: Could you add tests for strings (where Array.Clone()  should be used) and a non-primitive type (where Clone() shouldn't be used)?

**Review by CommonGuy — CHANGES_REQUESTED** [review]

File: src/Riok.Mapperly/Emit/SyntaxFactoryHelper.cs (hunk 1 above)
Comment (latonz) [reply src/Riok.Mapperly/Emit/SyntaxFactoryHelper.cs]: Removed.

File: test/Riok.Mapperly.Tests/Mapping/EnumerableTest.cs (hunk 2 above)
Comment (latonz) [reply test/Riok.Mapperly.Tests/Mapping/EnumerableTest.cs:13]: Added tests.

File: src/Riok.Mapperly/Emit/SyntaxFactoryHelper.cs [hunk 3]
```diff
@@ -141,17 +150,30 @@ public static InvocationExpressionSyntax StaticInvocation(IMethodSymbol method,
         return FieldDeclaration(variableDeclaration).WithModifiers(modifierTokenList);
     }
 
-    public static LocalDeclarationStatementSyntax DeclareVariable(string variableName, ExpressionSyntax initializationValue)
+    public static BinaryExpressionSyntax LessThan(ExpressionSyntax left, ExpressionSyntax right)
```
Comment (CommonGuy) [inline src/Riok.Mapperly/Emit/SyntaxFactoryHelper.cs:153]: Sorry, didn't catch this the last time. Seems unused.

**Review by CommonGuy — CHANGES_REQUESTED** [review]

File: src/Riok.Mapperly/Emit/SyntaxFactoryHelper.cs (hunk 3 above)
Comment (latonz) [reply src/Riok.Mapperly/Emit/SyntaxFactoryHelper.cs]: Oops removed.

**Review by CommonGuy — APPROVED** [review]

**Pull request merged**